### PR TITLE
IBP001 - Stair decomposition

### PR DESCRIPTION
**Pull request by Ghesselink**
Code found on my local machine that could serve as a best-practice

### CODE REVIEW

File: features/resources/spatial_DecompositionTable.csv [hunk 1]
```diff
@@ -43,4 +43,8 @@ IfcSite,IfcRailway
 IfcSite,IfcRoad
 IfcSite,IfcExternalSpatialElement
 IfcSite,IfcSpace
-IfcSpace,IfcSpace
\ No newline at end of file
+IfcSpace,IfcSpace
```
Comment (aothms) [inline features/resources/spatial_DecompositionTable.csv:46]: I think this should be a separate table. IfcStair is not a spatial element. Or the table can be renamed to simply decomposition table. But people are going to complain if we don't have consistent vocabulary on these kind of things.

File: test/files/ibp001/generate.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import os
+import ifcopenshell
+import ifcopenshell.template
+
+for validity, elem in [
+    ("pass", 'IfcRailing'),
+    ("pass", 'IfcSlab'),
+    ("pass", 'IfcStairFlight'),
+    ("pass", None),
+    ("fail", "IfcBeam"),
```
Comment (aothms) [inline test/files/ibp001/generate.py:10]: I would say a Beam should/could be allowed, same as an IfcMember.

**Review by aothms — CHANGES_REQUESTED** [review]

**Comment by evandroAlfieri** [comment]
> I'm not immediately falling in love with the rule code ASS101, but maybe I'm being too conservative.

I expect @Ghesselink naming convention script to flag it. And not because it's conservative too...but because the feature title, i.e. ASS101, is inconsistent with the file name, i.e. IBP001 😄

**Comment by Ghesselink** [comment]
I changed it too IPB (Industry Best Practice) for the same reason 😛

File: test/files/ibp001/generate.py (hunk 2 above)
Comment (Ghesselink) [reply test/files/ibp001/generate.py]: And what about IfcRoof and IfcWall, for instance? Should they be allowed? 

The documentation I found about it only mention flights, railings and slabs. e.g.; https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcStair.htm

File: features/resources/spatial_DecompositionTable.csv (hunk 1 above)
Comment (Ghesselink) [reply features/resources/spatial_DecompositionTable.csv]: Moved it to a separate table for now

**Review by evandroAlfieri — APPROVED** [review]
Nice. Let's merge!

**Pull request merged**